File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/items/equipment/BodyEquipment.java
package de.limbusdev.guardianmonsters.guardians.items.equipment;


import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian;

/**
 * BodyEquipment
 *
 * @author Georg Eckert 2017
 */

public class BodyEquipment extends Equipment {

    public enum Type {
        ARMOR, BARDING, SHIELD, BREASTPLATE,
    }

    private Type type;

    public BodyEquipment(String name, Type type, int PStr, int PDef, int MStr, int MDef, int Speed, int HP, int MP, int Exp) {
        super(name, BodyPart.BODY, PStr, PDef, MStr, MDef, Speed, HP, MP, Exp);
        this.type = type;
    }

    public Type getType() {
        return type;
    }

    @Override
    public boolean equipable(AGuardian m) {
        if(super.equipable(m)) {
            return (m.getSpeciesDescription().getBodyType() == type);
        } else {
            return false;
        }
    }
}

File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/items/equipment/BodyEquipment.kt
package de.limbusdev.guardianmonsters.guardians.items.equipment


import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian

/**
 * BodyEquipment
 *
 * @author Georg Eckert 2017
 */

class BodyEquipment(

        name: String,
        val type: Type,
        PStr: Int,
        PDef: Int,
        MStr: Int,
        MDef: Int,
        Speed: Int,
        HP: Int,
        MP: Int,
        Exp: Int

) : Equipment(name, BodyPart.BODY, PStr, PDef, MStr, MDef, Speed, HP, MP, Exp)
{
    override fun canBeEquipped(guardian: AGuardian): Boolean
    {
        return super.canBeEquipped(guardian) && guardian.speciesDescription.bodyType == type
    }


    // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ Inner Classes
    enum class Type
    {
        ARMOR, BARDING, SHIELD, BREASTPLATE
    }
}